[quant][graphmode] Add support for functional linear

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40384 [quant][graphmode][refactor] InsertQuantDeQuantHelper
* #40383 [quant][graphmode] Add warning for debug option for add_scalar/mul_scalar
* **#40331 [quant][graphmode] Add support for functional linear**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22162905](https://our.internmc.facebook.com/intern/diff/D22162905)

jerryzh168 committed:
[quant][graphmode] Add support for functional linear

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add support for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add support for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add support for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add support for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@z-a-f this should fix the problem you see when testing traced linear_relu

jerryzh168 committed:
Update on "[quant][graphmode] Add support for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add support for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add support for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add support for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on torch/csrc/jit/passes/quantization/insert_observers.cpp:
```diff
@@ -476,31 +476,34 @@ class InsertObserversHelper {
   const PatternInfo nn_linear_nn_relu = PatternInfo::parse_from_str(
       R"(
-graph(%self, %linear, %relu, %input):
+graph(%input, %linear, %relu):
```
For my understanding, why was self part of the graph, when it is not used in the actual pattern?

jerryzh168 replied:
yeah, that's why I removed it, it's not needed in the pattern.

jerryzh168 committed:
Update on "[quant][graphmode] Add support for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22162905](https://our.internmc.facebook.com/intern/diff/D22162905)

[ghstack-poisoned]

vkuzo reviewed (approved):
lg

jerryzh168 committed:
Update on "[quant][graphmode] Add support for functional linear"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22162905](https://our.internmc.facebook.com/intern/diff/D22162905)

[ghstack-poisoned]
